kuwala/core/neo4j/importer/src/PipelineConnector.py

```diff
@@ -33,40 +42,76 @@ def connect_h3_indexes():
 
     Neo4jConnection.close_connection()
 
+    end_time = time.time()
+
+    print(f'Connected H3 indexes in {round(end_time - start_time)} s')
+
+
+def add_pois(df):
+    query = '''
+        MERGE (p:Poi { id: event.id })
+        WITH event, p
+        MATCH (h:H3Index { h3Index: event.h3Index })
+        MERGE (p)-[:LOCATED_AT]->(h)
+    '''
+
+    Neo4jConnection.spark_send_query(df, query)
+
+
+def connect_osm_pois(df):
+    query = '''
+        MATCH (p:Poi { id: event.id })
+        WITH event, p
+        MATCH (po:PoiOSM { id: event.osmId })
+        MERGE (po)-[:BELONGS_TO]->(p)
+    '''
+
+    Neo4jConnection.spark_send_query(df, query)
+
+
+def connect_google_pois(df):
+    query = '''
+        MATCH (p:Poi { id: event.id })
+        WITH event, p
+        MATCH (pg:PoiGoogle { id: event.googleId })
+        MERGE (pg)-[:BELONGS_TO { confidence: event.confidence }]->(p)
+    '''
+
+    Neo4jConnection.spark_send_query(df, query)
+
 
 # Create one single POI node combining OSM and Google
 def connect_pois(df_osm: DataFrame, df_google: DataFrame):
-    df_osm = df_osm.select('id', 'h3Index').withColumnRenamed('h3Index', 'h3IndexOsm').withColumnRenamed('id', 'osmId')
+    start_time = time.time()
+    df_osm = df_osm.select('id', 'h3_index') \
+        .withColumnRenamed('h3_index', 'h3IndexOsm') \
+        .withColumnRenamed('id', 'osmId')
     df_google = df_google \
         .withColumn('osmId', df_google['osmId'].cast(LongType())) \
         .withColumn('osmId', concat('type', 'osmId')) \
         .select('id', 'osmId', 'confidence', 'h3Index') \
         .withColumnRenamed('id', 'googleId') \
         .withColumnRenamed('h3Index', 'h3IndexGoogle')
-    df_pois = df_osm.join(df_google, on=['osmId'], how='left')
 
-    Neo4jConnection.connect_to_graph()
-    Neo4jConnection.query_graph('CREATE CONSTRAINT poi IF NOT EXISTS ON (p:Poi) ASSERT p.id IS UNIQUE')
+    @udf(returnType=StringType())
+    def get_poi_id(confidence, h3_index_google, h3_index_osm, osm_id):
```

Comment:
Better naming here could be more future proof, get_poi_id does not tell me much about what this function does. build_poi_id_based_on_confidence ?